src/librustc_mir/borrow_check/error_reporting.rs

```diff
@@ -538,6 +574,66 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
         err
     }
 
+    fn find_name_and_ty_of_values(
+        &self,
+    ) -> Option<(String, Span, String, Span)> {
+        let mir_node_id = self.infcx.tcx.hir.as_local_node_id(self.mir_def_id)?;
+        let fn_decl = self.infcx.tcx.hir.fn_decl(mir_node_id)?;
+
+        // If there is one argument and this error is being reported, that means
+        // that the lifetime of the borrow could not be made to match the single
+        // argument's lifetime. We can highlight it.
+        //
+        // If there is more than one argument and this error is being reported, that
+        // means there must be a self parameter - as otherwise there would be an error
+        // from lifetime elision and not this. So we highlight the self parameter.
```

Comment:
I don't follow this logic 100% — this is true if the region in question is anonymous, but that's not known here, right?

Reply:
And this only looks at the first lifetime?

Reply:
I've updated the logic to handle this case correctly.